src/PostgREST/Query/QueryBuilder.hs

```diff
@@ -133,7 +133,9 @@ mutatePlanToQuery (Update mainQi uCols body logicForest range ordts returnings)
     SQL.sql (returningF mainQi returnings)
 
   where
-    whereLogic = if null logicForest then mempty else " WHERE " <> intercalateSnippet " AND " (pgFmtLogicTree mainQi <$> logicForest)
+    whereLogic = if null logicForestF && (not isBulk || null pkFlts) then mempty else " WHERE " <> intercalateSnippet " AND " (logicForestF <> pgrstUpdateBodyF)
+    logicForestF = pgFmtLogicTree mainQi <$> logicForest
+    pgrstUpdateBodyF = if not isBulk then mempty else pgFmtBodyFilter mainQi (QualifiedIdentifier mempty "pgrst_body") <$> pkFlts
```

Comment:
```suggestion
    whereLogic = 
      if null logicForestF && (not isBulk || null pkFlts) 
        then mempty 
        else " WHERE " <> intercalateSnippet " AND " (logicForestF <> pgrstUpdateBodyF)
    logicForestF = pgFmtLogicTree mainQi <$> logicForest
    pgrstUpdateBodyF = 
      if not isBulk 
         then mempty 
         else pgFmtBodyFilter mainQi (QualifiedIdentifier mempty "pgrst_body") <$> pkFlts
```

Reply:
Also, would it be possible to refactor this to only execute the `not isBulk` conditional once? That would be clearer I think.